### DISCUSSION TITLE
[QUESTION] H265 support needed

### DISCUSSION THREAD

**Discussion by gijsbeijer** [awslabs/amazon-kinesis-video-streams-producer-sdk-cpp#1106] [Ideas]
It would be great if the examples would contain samples that would work for h265 streams as well (for me personally the rtsp multistream would be great).

I've been trying to change the existing code to get h265 supported but without any luck. Below is the log of the part where it shows that theres a mismatch in content-types.

```
ocuma-rtsp_stream-1  | [INFO ] [17-11-2023 18:42:02:943.312 GMT] describeStreamResultEvent(): Describe stream result event.
ocuma-rtsp_stream-1  | [WARN ] [17-11-2023 18:42:02:943.598 GMT] describeStreamResult(): Content type returned from the DescribeStream(video/h264) call doesn't match the one specified in the StreamInfo(video/x-h265)
ocuma-rtsp_stream-1  | [WARN ] [17-11-2023 18:42:02:943.750 GMT] describeStreamResult(): [Videostream_0] Content type returned from the DescribeStream call doesn't match the one specified in the StreamInfo
ocuma-rtsp_stream-1  | [INFO ] [17-11-2023 18:42:03:135.119 GMT] writeHeaderCallback(): RequestId: b0405bc5-fd81-43c3-a3f8-423efd1dbb37
ocuma-rtsp_stream-1  | [DEBUG] [17-11-2023 18:42:03:135.567 GMT] getStreamingEndpointCurlHandler(): [Videostream_0] GetStreamingEndpoint API response: {"DataEndpoint":"https://s-96607c7a.kinesisvideo.eu-central-1.amazonaws.com"}
ocuma-rtsp_stream-1  | [INFO ] [17-11-2023 18:42:03:138.965 GMT] getStreamingEndpointResultEvent(): Get streaming endpoint result event.
ocuma-rtsp_stream-1  | [DEBUG] [17-11-2023 18:42:03:139.302 GMT] getStreamingTokenHandler invoked
ocuma-rtsp_stream-1  | [DEBUG] [17-11-2023 18:42:03:139.391 GMT] Refreshing credentials. Force refreshing: 1 Now time is: 1700246523139379187 Expiration: 1700250122
ocuma-rtsp_stream-1  | [INFO ] [17-11-2023 18:42:03:139.543 GMT] New credentials expiration is 1700250123
ocuma-rtsp_stream-1  | [INFO ] [17-11-2023 18:42:03:139.657 GMT] getStreamingTokenResultEvent(): Get streaming token result event.
ocuma-rtsp_stream-1  | [DEBUG] [17-11-2023 18:42:03:140.235 GMT] streamReadyHandler invoked
ocuma-rtsp_stream-1  | [DEBUG] [17-11-2023 18:42:03:140.900 GMT] Stream is ready: Videostream_0
```

Does anyone have an h265 (multi-)stream running??

**Comment by hassanctech** [comment]
The Producer C SDK (the CPP SDK is a thin wrapper over the C SDK) contains samples for h265, here's the README for how to use h265 (there are sample h265 frames provided as well):  https://github.com/awslabs/amazon-kinesis-video-streams-producer-c/blob/master/README.md#run-samples

You can look there and try to copy over the differences between the H264 and H265 variant in the C Producer SDK to your use case for producer sdk c++.  It seems your question is requesting a sample update in the C++ SDK to support. I will mark this as an enhancement and move this to a discussion as it's not an open question / bug.

**Reply by gijsbeijer** [reply]
Hi I tried to get it to run with adjusting the cpp files according to the C version but as I am not a native C++ or C dev I'm probably missing something.

If we could get a sample with h265 running that would be great!
